### PR TITLE
vmclone: improve test time

### PR DESCRIPTION
**Pull request by fossedihelm**
### What this PR does
This series of commits gradually remove complexity of vmclone tests to the point where we reduce the runtime from ~3 seconds down to ~250ms. On top of this we remove some unnecessary code and fix few issues.

### Special notes for your reviewer
These changes follow the logic introduced here https://github.com/kubevirt/kubevirt/pull/12468

### Release note

```release-note
NONE
```

### CODE REVIEW

**Comment by fossedihelm** [comment]
/hold
Based on https://github.com/kubevirt/kubevirt/pull/12219

File: pkg/virt-controller/watch/clone/clone_test.go [hunk 1]
```diff
@@ -61,18 +61,16 @@ const (
 
 var _ = Describe("Clone", func() {
 	var (
-		ctrl        *gomock.Controller
```
Comment (xpivarc) [inline pkg/virt-controller/watch/clone/clone_test.go]: How do we achieve "Remove tests/util pkg dependency " in e7dae50d1ba97296eeb8b5427849c24b4c2b23ab ?

**Review by xpivarc — COMMENTED** [review]
/approve
/hold
I think is not clear to me

File: pkg/virt-controller/watch/clone/clone_test.go (hunk 1 above)
Comment (fossedihelm) [reply pkg/virt-controller/watch/clone/clone_test.go]: Oh... I had already handled it in https://github.com/kubevirt/kubevirt/commit/087f26e92691f2e8a7c18be38f0f79f405340297 ad during rebase I haven't noticed it. Will reword the commit message. Thank you!

Comment (fossedihelm) [reply pkg/virt-controller/watch/clone/clone_test.go]: Done!

**Comment by acardace** [comment]
/lgtm

very nice!

**Pull request merged**